filecoin-proofs/src/api/internal.rs

```diff
@@ -200,24 +212,11 @@ pub struct PoStInput {
     pub input_parts: Vec<PoStInputPart>,
 }
 
-pub fn fake_generate_post(_sector_bytes: u64, input: PoStInput) -> error::Result<PoStOutput> {
-    let faults: Vec<u64> = if !input.input_parts.is_empty() {
-        vec![0]
-    } else {
-        Default::default()
-    };
-
-    Ok(PoStOutput {
-        snark_proof: [42; 192],
-        faults,
-    })
-}
-
-pub fn generate_post(sector_bytes: u64, input: PoStInput) -> error::Result<PoStOutput> {
+pub fn generate_post(sector_config: &SectorConfig, input: PoStInput) -> error::Result<PoStOutput> {
     let faults: Vec<u64> = Vec::new();
 
     let setup_params = compound_proof::SetupParams {
-        vanilla_params: &post_setup_params(sector_bytes as usize),
+        vanilla_params: &post_setup_params(input.input_parts.len(), sector_config.sector_bytes()),
```

Comment:
If I'm following this correctly, the number of sectors is the number of `PoStInputPart`s, right?